src/persist-txn/src/lib.rs

```diff
@@ -294,16 +294,16 @@ pub mod txns;
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum TxnsEntry {
     /// A data shard register operation.
-    Register(ShardId),
+    Register(ShardId, [u8; 8]),
     /// A batch written to a data shard in a txn.
-    Append(ShardId, Vec<u8>),
+    Append(ShardId, [u8; 8], Vec<u8>),
```

Comment:
Might be useful to indicate what the `[u8; 8]` represents.

Reply:
Good call, done!